core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,1092 @@
+  }
+  public static create(
+    context: EllipticalArcApproximationContext, options: EllipticalArcApproximationOptions,
+  ): NonUniformCurvatureSampler {
+    return new NonUniformCurvatureSampler(context, options);
+  }
+  /**
+   * Compute the angle corresponding to the point in the ellipse's first quadrant with the given curvature.
+   * * The elliptical arc is assumed to be non-circular and have perpendicular axes of positive length; its sweep is ignored.
+   * * This is a scaled inverse of [[Arc3d.fractionToCurvature]] restricted to fractions in [0, 1/4].
+   * @return radian angle in [0, pi/2] or undefined if the ellipse is invalid, or does not attain the given curvature.
+   */
+  private curvatureToRadians(curvature: number): number | undefined {
+    /*
+    Let the elliptical arc be parameterized with axes u,v of different length and u.v = 0:
+      f(t) = c + u cos(t) + v sin(t),
+      f'(t) = -u sin(t) + v cos(t),
+      f"(t) = -u cos(t) - v sin(t)
+    We seek a formula for t(K), the inverse of the standard curvature formula
+      K(t) := ||f'(t) x f"(t)||/||f'(t)||^3
+    for a parametric function f(t):R->R^3. We'll restrict K to Q1 (i.e., t in [0, pi/2]), where K is monotonic.
+    By linearity of the cross product and the above formulas, the numerator of K(t) reduces to ||u x v||, and so:
+      cbrt(||u x v||/K) = ||f'(t)|| = sqrt(f'(t).f'(t))
+    Leveraging u,v perpendicularity we can define:
+      lambda(K) := (||u x v||/K)^(2/3) = (||u|| ||v|| / K)^(2/3) = cbrt(u.u v.v / K^2)
+    Then substituting and using perpendicularity again:
+      lambda(K) = f'(t).f'(t)
+        = sin^2(t)u.u + cos^2(t)v.v - 2sin(t)cos(t)u.v
+        = u.u + cos^2(t)(v.v - u.u)
+    Taking the positive root because cos(t)>=0 in Q1, and relying on u,v having different lengths:
+      cos(t) = sqrt((lambda(K) - u.u)/(v.v - u.u))
+    Solving for t yields the formula for t(K).
+    */
+    if (!this._curvatureRange.containsX(curvature))
+      return undefined; // ellipse does not attain this curvature
+    const lambda = Math.cbrt((this._xMag2 * this._yMag2) / (curvature * curvature));
+    const cosTheta = Math.sqrt(Math.abs((lambda - this._xMag2) / (this._yMag2 - this._xMag2)));
+    return Math.acos(cosTheta);
+  }
+  public computeRadiansStrictlyInsideQuadrant1(result?: number[]): number[] {
+    if (!result)
+      result = [];
+    if (this._context.isValidArc) {
+      const tDelta = 1.0 / (this._options.numSamplesInQuadrant - 1);
+      for (let i = 1; i < this._options.numSamplesInQuadrant - 1; ++i) {
+        const j = this._options.remapFunction(i * tDelta);
+        const curvature = (1 - j) * this._curvatureRange.low + j * this._curvatureRange.high;
+        const angle = this.curvatureToRadians(curvature);
+        if (undefined !== angle)
+          result.push(angle);
+      }
+    }
+    return result;
+  }
+};
+
+/**
+ * Implementation for method `EllipticalArcSampleMethod.UniformCurvature`
```

Comment:
Maybe in the doc say something about this class basically uses non unifrom curvature with identity map which makes it unifrom!